### ISSUE TITLE
[monocdk] Unable to create custom stacks that extend Stack

### ISSUE THREAD

**Issue by agdimech** [aws/aws-cdk#11362]
When trying to synthesize a custom Stack (which extends Stack), I receive the below error:

"Class constructor Stack cannot be invoked without 'new'"

Here is the erroneous sample code:

```
export class FooStack extends Stack {
    constructor(construct: Construct) {
        super(construct);

        new MyCustomConstruct(construct, "test");
    }

}

const app = new App();
const stack = new FooStack(app);
app.synth();
```

However if using the Stack directly it works just fine as per the below sample code:

```
const app = new App();
const stack = new Stack(app, "testStack");
new SecureBucket(stack, 'test');
app.synth();
```

I am using the Construct from monocdk, tried with the Construct from the constructs package however received a different error relating to cannot find children.

### Reproduction Steps

```
export class FooStack extends Stack {
    constructor(construct: Construct) {
        super(construct);

        new MyCustomConstruct(construct, "test");
    }

}

const app = new App();
const stack = new FooStack(app);
app.synth();
```

### What did you expect to happen?

Stack should syntesize.

### What actually happened?

Class constructor Stack cannot be invoked without 'new'

### Environment

  - **CDK CLI Version  : 1.71.0 (build 953bc25)**
  - **Framework Version:   "constructs": "3.0.4", "monocdk": "1.69.0"**
  - **Node.js Version: v14.14.0** 
  - **OS               : OS/X Catalina**
  - **Language (Version): Typescript 3.9.5**

--- 

This is :bug: Bug Report

**Comment by NetaNir** [comment]
You should be using `Construct` from the `constructs`  module. Could you add the error you see when using the `construct` model?

**Comment by agdimech** [comment]
> You should be using `Construct` from the `constructs` module. Could you add the error you see when using the `construct` module?

monocdk/lib/core/lib/private/synthesis.ts:40
    for (const child of root.node.children) {
                                  ^
TypeError: Cannot read property 'children' of undefined
    at synthNestedAssemblies (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/private/synthesis.ts:40:35)
    at synthNestedAssemblies (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/private/synthesis.ts:45:13)
    at synthNestedAssemblies (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/private/synthesis.ts:45:13)
    at Object.synthesize (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/private/synthesis.ts:13:5)
    at App.synth (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/stage.ts:160:29)
    at process.<anonymous> (/Users/dimecha/Projects/AWSPDKConstructs/src/AWSPDKConstructs/node_modules/monocdk/lib/core/lib/app.ts:109:51)
    at Object.onceWrapper (events.js:422:26)
    at process.emit (events.js:315:20)
    at process.EventEmitter.emit (domain.js:486:12)
    at process.emit (/Users/dimecha/.npm/_npx/63517/lib/node_modules/ts-node/node_modules/source-map-support/source-map-support.js:495:21)

**Comment by agdimech** [comment]
Here is some sample code:

```
import {App, Stack} from 'monocdk';
import {Construct} from 'constructs';
import {Bucket} from 'monocdk/aws-s3';

export class FooStack extends Stack {
    constructor(construct: Construct) {
        super(construct);

        new Bucket(construct, "test");
    }

}

const app = new App();
const stack = new FooStack(app);
app.synth();
```

With error: Class constructor Stack cannot be invoked without 'new'

And similarly:

```
import {App, Stack} from 'monocdk';
import {Construct} from 'constructs';
import {Bucket} from 'monocdk/aws-s3';

export class BucketWrapper extends Construct {
    constructor(scope: Construct, id: string) {
        super(scope, id);
        new Bucket(this, 'SomeBucket');
    }
}

const app = new App();
const stack = new Stack(app);
new BucketWrapper(stack, 'test');
app.synth();
```

With error: Class constructor Stack cannot be invoked without 'new'

**Comment by NetaNir** [comment]
This usually happens when your tsconfig file have an older version of ECMAScript configured. For reference you can check out [delivlib](https://github.com/awslabs/aws-delivlib/blob/a367f0f4f185b9ca1b4f43a797a86a3ad3a1f3b3/pipeline/delivlib.ts) which also uses monocdk (it uses monocdk-experiment which is the older version of monocdk).

Also note that monocdk is experimental and should not be used for production applications. It is mainly released publicly for us to experiment with it.

**Comment by blimmer** [comment]
In case someone else finds this issue searching for the `Class constructor Stack cannot be invoked without 'new'` error message, I ran into this problem with an internal, shared CDK library using [`tsdx`](https://tsdx.io/) to bundle. By default, it bundles for a `web` target, which was not exposing true Javascript `class`es in the output.

To fix it, I needed to use `tsdx build --target node`. Hopefully this helps someone else!
